src/cli/list.rs

```diff
@@ -0,0 +1,182 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use rattler_conda_types::Platform;
+use rattler_lock::LockedDependencyKind;
+use serde::Serialize;
+
+use crate::lock_file::load_lock_file;
+use crate::project::SpecType;
+use crate::Project;
+
+/// List project's packages. Highlighted packages are explicit dependencies.
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = false)]
+pub struct Args {
+    /// The platform to list packages for. Defaults to the current platform.
+    #[arg(long)]
+    pub platform: Option<Platform>,
+
+    /// Whether to output in json format
+    #[arg(long)]
+    pub json: bool,
+
+    /// Whether to output in pretty json format
+    #[arg(long)]
+    pub json_pretty: bool,
+
+    /// Whether to sort the package list by name
+    #[arg(long)]
+    pub no_sort: bool,
+
+    /// The path to 'pixi.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+}
+
+#[derive(Serialize)]
+struct PackageToOutput {
+    name: String,
+    version: String,
+    build: Option<String>,
+    kind: String,
+    channel: Option<String>,
+    is_explicit: bool,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    // Load the project
+    let project = Project::load_or_else_discover(args.manifest_path.as_deref())
+        .ok()
+        .ok_or_else(|| miette::miette!("No project found"))?;
+
+    // Load the platform
+    let platform = args.platform.unwrap_or_else(Platform::current);
+
+    // Load the environment
+    // NOTE(hadim): make it an argument once environments are implemented.
+    // let environment = project.default_environment();
+
+    // Load the lockfile
+    let lockfile = load_lock_file(&project)
+        .await
+        .map_err(|_| miette::miette!("Cannot load lockfile. Did you run `pixi install` first?"))?;
+
+    let locked_deps = lockfile.packages_for_platform(platform).collect::<Vec<_>>();
+
+    // Get the explicit project dependencies
+    let project_dependency_names: Vec<String> = {
+        let dependencies = project
+            .default_environment()
+            .dependencies(Some(SpecType::Run), Some(platform));
+        dependencies
+            .names()
+            .map(|p| p.as_source().to_string())
+            .collect()
+    };
+
+    // Convert the the list of package record to specific output format
+    let mut packages_to_output = locked_deps
+        .iter()
+        .map(|p| {
+            match p.kind {
+                // Conda package
+                LockedDependencyKind::Conda(_) => {
+                    let name = p.name.clone();
+                    let version = p.version.clone();
+                    let kind = "conda".to_string();
+                    let build = p.as_conda().unwrap().build.clone();
+                    // NOTE(hadim): channel is not available, only the source url...
+                    // let channel = Some(p.as_conda().unwrap().url.to_string().clone());
```

Comment:
I am now loading the packages from the lock file, so I can specify the platform to list (and the env in the future).

But I cannot find the way to retrieve the conda channel associated to a conda lock dep. Any idea?

(besides parsing the source url)